cmd/utils.go

```diff
@@ -314,3 +317,16 @@ func ceilFrac(numerator, denominator int64) (ceil int64) {
 	}
 	return
 }
+
+func newContext(r *http.Request, api string) context.Context {
+	vars := mux.Vars(r)
+	bucket := vars["bucket"]
+	object := vars["object"]
+	prefix := vars["prefix"]
+
+	if prefix != "" {
+		object = prefix
```

Comment:
not only mux.Vars() we also should handle the query parameter values that get extracted in ListObjects() operation.

Reply:
yes @harshavardhana we pushed this PR as a base framework. We can iteratively improve the details. For ex the other query parameters can be added to ReqInfo.Tags by ListObjects call in future. Because getting everything right in a single big PR will be hard as we will have to keep up with conflicts with the master.